crates/apub/src/protocol/objects/page.rs

```diff
@@ -124,6 +125,40 @@ impl Attachment {
       _ => None,
     }
   }
+
+  pub(crate) fn as_markdown(&self) -> String {
+    match self {
+      Attachment::Image(i) => {
+        format!(
+          "![{}]({})",
+          i.name.as_ref().unwrap_or(&String::default()),
+          i.url
+        )
+      }
+      Attachment::Document(d) => {
+        if let Some(media) = &d.media_type {
+          if ["video/", "image/"].iter().any(|s| media.starts_with(s)) {
```

Comment:
Any reason youre doing link formatting only for certain content types? Would make sense to do it always, because some frontends may not convert plain urls to clickable links.

Also you can simply return the url from the match, and do format after that, to avoid duplicate code.